### ISSUE TITLE
Cache pulled docker images

### ISSUE THREAD

**Issue by rahulbirari-cpi** [actions/cache#712]
Hi I have been using git actions to build for embedded code with docker container which are already build and pushed to docker-hub.

Every time the build is triggered docker image (~5Gbs) is downloaded from Docker hub and job is processed in in a runner, which consumes about 4-5 minutes of time for the job 50% of which is a build time.

All I want to know is is there any way to cache these 'pulled' images in git actions **efficiently** ? 
I tried **docker save** and **docker load** along with **actions/cache** but processing time for the .tar file is huge as compared to downloading the image. 

Moreover, if cached docker image can be loaded quickly, build can be significantly parallelized and can save upto 80% of total time.

In summary : Is there any way to cache **pulled docker** image **instead of** downloading it for every job and building it on the git actions ?

**Comment by bishal-pdMSFT** [comment]
@rahulbirari-cpi cache actions does not explicitly support docker image caching, but there is a docker [buildx](https://github.com/marketplace/actions/docker-setup-buildx) action which supports [caching](https://github.com/moby/buildkit#github-actions-cache-experimental) (and has an option to use GitHub backed cache)
